docs/jwst/skymatch/description.rst

```diff
@@ -0,0 +1,239 @@
+
+``usig`` (float, default=4.0)
+  Upper clipping limit, in sigma, used when computing the sky value.
+
+``binwidth`` (float, default=0.1)
+  Bin width, in sigma, used to sample the distribution of pixel
+  values in order to compute the sky background using statistics
+  that require binning, such as `mode` and `midpt`.
+
+
+Limitations and Discussions
+---------------------------
+The primary reason for introducing the ``skymatch`` algorithm was to try to
+equalize the sky in large mosaics in which computation of the
+absolute sky is difficult, due to the presence of large diffuse
+sources in the image. As discussed above, the ``skymatch`` step
+accomplishes this by comparing the sky values in the
+overlap regions of each image pair. The quality of sky matching will
+obviously depend on how well these sky values can be estimated.
+True background may not be present at all in some images, in which case
+the computed "sky" may be the surface brightness of a large galaxy, nebula, etc.
+
+Here is a brief list of possible limitations and factors that can affect
+the outcome of the matching (sky subtraction in general) algorithm:
+
+- Because sky computation is performed on *flat-fielded* but
+  *not distortion corrected* images, it is important to keep in mind
+  that flat-fielding is performed to obtain correct surface brightnesses.
+  Because the surface brightness of a pixel containing a point-like source will
+  change inversely with a change to the pixel area, it is advisable to
+  mask point-like sources through user-supplied mask files. Values
+  different from zero in user-supplied masks indicate good data pixels.
+  Alternatively, one can use the ``upper`` parameter to exclude the use of
+  pixels containing bright objects when performing the sky computations.
+
+- The input images may contain cosmic rays. This
+  algorithm does not perform CR cleaning. A possible way of minimizing
+  the effect of the cosmic rays on sky computations is to use
+  clipping (\ ``nclip`` > 0) and/or set the ``upper`` parameter to a value
+  larger than most of the sky background (or extended sources) but
+  lower than the values of most CR-affected pixels.
+
+- In general, clipping is a good way of eliminating bad pixels:
+  pixels affected by CR, hot/dead pixels, etc. However, for
+  images with complicated backgrounds (extended galaxies, nebulae,
+  etc.), affected by CR and noise, the clipping process may mask different
+  pixels in different images. If variations in the background are
+  too strong, clipping may converge to different sky values in
+  different images even when factoring in the true difference
+  in the sky background between the two images.
+
+- In general images can have different true background values
+  (we could measure it if images were not affected by large diffuse
+  sources). However, arguments such as ``lower`` and ``upper`` will
+  apply to all images regardless of the intrinsic differences
+  in sky levels.
+
```

Comment:
I would add what happens if you have a mosaic with images not all overlapping. For example, if you have two groups of images not overlapping with each other, what does 'skymatch' do? Does it work as if the two groups are two distinct mosaic? I think this is particularly important when 'match' is used

Reply:
Good question. I'll check into that.

Reply:
I've done the experiment of having gaps on the sky and have included it in the latest examples. Bottom line: gaps are not good for the match method.